source/glimmer-components/passing-data-to-components.md

```diff
@@ -0,0 +1,231 @@
+Imagine if functions aren't allowed to take arguments: they won't be very
+useful at all, right? Likewise, if we cannot pass data to our Components, they
+would just be fancy partials with limited uses. Fortunately, Ember made it very
+easy to pass data into our Components.
+
+## Passing Attributes
+
+When invoking Components, you can pass along any number of named attributes,
+just like how you can set attributes on regular HTML elements:
+
+```app/templates/contact-us.hbs
+<h5><x-icon type="speech-bubble" size="large" /> Contact Us</h5>
+
+<x-icon type="phone" /> 1-800-EMBER-JS <br>
+<x-icon type="twitter" /> @emberjs <br>
+<x-icon type="map" /> 123 Some Street, Portland OR
+```
+
+These attributes can then be accessed from the Component's template via the
+`attrs` hash:
+
+```app/templates/components/x-icon.hbs
+<x-icon class="icon icon-{{attrs.type}} {{attrs.size}}" />
+```
+
+You might be wondering: what's the point of going through the `attrs` hash?
+Couldn't Ember just expose the attributes as `{{type}}` instead of the nosier
+`{{attrs.type}}`? Hold your thoughts! We will come back to that soon.
+
+In addition to string attributes, sometimes you might need to pass other kind
+of data such as numbers and booleans. You can do this using the curlies syntax
+instead of quotation marks:
+
+```app/templates/step-two.hbs
+<progress-bar value={{2}} max={{10}} animated={{true}} /> Step 2 / 10
```

Comment:
The only other divergence from HTML is:

```hbs
<input checked={{isActive}}>
```

In HTML, attributes without quotes are still attributes, but in Ember, the above syntax assigns a property, not an attribute.

I think we can get away with this divergence because we are introducing new syntax that allows you to add dynamic content to HTML (`{{}}` is pretty much the consensus syntax for this extension). Because assigning both attributes and properties to HTML (especially with web components) is common, having a terse syntax for describing each is important.

Again, this restricts the divergence from "normal" HTML to:

* `/>` means what it means in XHTML
* `foo={{}}` without quotation marks around the curlies means assigning a property, not an attribute.